core/domain/learner_group_services.py

```diff
@@ -494,6 +504,46 @@ def add_student_to_learner_group(
     learner_group_model.put()
 
 
+def remove_students_from_learner_group(
+    group_id: str,
+    user_ids: List[str]
+) -> None:
+    """Removes the given student from the given learner group.
+
+    Args:
+        group_id: str. The id of the learner group.
+        user_ids: List[str]. The id of the students to be removed.
+    """
+    learner_group_model = learner_group_models.LearnerGroupModel.get(
+        group_id, strict=True
+    )
```

Comment:
This seems like an unnecessary database fetch. Can't the caller function pass the learner group domain object?

Reply:
the model needs to be updated so the domain object won't do. But have added conditional fetching only when the model needs to be updated.